test/fake-test.js

```diff
@@ -160,25 +160,42 @@ describe("fake", function () {
     describe(".yields", function () {
         verifyProxy(fake.yields, noop, "42", "43");
 
-        it("should call the callback with the provided values", function () {
+        it("should call a callback with the provided values", function () {
             var callback = sinon.spy();
-            var myFake = fake.yields(callback, "one", "two", "three");
+            var myFake = fake.yields("one", "two", "three");
 
-            myFake();
+            myFake(callback);
 
             sinon.assert.calledOnce(callback);
             sinon.assert.calledWith(callback, "one", "two", "three");
         });
+
+        it("should call the last function argument", function () {
+            var callback = sinon.spy();
+            var myFake = fake.yields();
+
+            myFake(function () {}, callback);
+
+            sinon.assert.calledOnce(callback);
+        });
+
+        it("should throw if the last argument is not a function", function () {
+            var myFake = fake.yields();
+
+            assert.exception(function () {
+                myFake(function () {}, "not a function");
+            }, /TypeError: Expected last argument to be a function/);
```

Comment:
Does [`assert.exception`](http://docs.busterjs.org/en/latest/modules/referee/#exception) work as expected with regular expressions for the matcher argument?

Reply:
Yes